File: test/py/test_phyllochron.py
'Test generation'

from phyllochron import *
from math import *
import numpy as np



def test_test_wheat1():
    params= phyllochron(
    pincr = 1.25,
    gai = 0.279874189539498,
    ldecr = 3,
    fixPhyll = 91.2,
    lincr = 8,
    pdecr = 0.4,
     )
    phyllochron_estimated = round(params[0], 2)
    phyllochron_computed = 36.48
    assert (phyllochron_estimated == phyllochron_computed)
    pastMaxAI_estimated = round(params[1], 2)
    pastMaxAI_computed = 0
    assert (pastMaxAI_estimated == pastMaxAI_computed)

File: test/java/test_Phyllochron.java
//Test generation'
package pheno2;  
import java.math.BigDecimal;
class test_Phyllochron
{
    //check wheat model);
    //test_wheat1);

    @SuppressWarnings("empty-statement")
    public static void test_wheat1()
    {
        double fixPhyll = 91.2;
        double leafNumber = 0;
        double lincr = 8;
        double ldecr = 3;
        double pdecr = 0.4;
        double pincr = 1.25;
        double lai = 0;
        double ptq = 0;
        double gai = 0.279874189539498;
        double pastMaxAI = 0;
        double kl = 0.45;
        double aPTQ = 0.842934;
        double phylPTQ1 = 20;
        double p = 120;
        String choosePhyllUse = "Default";

        Phyllochron res0 = Estimation_Phyllochron.CalculatePhyllochron(fixPhyll,leafNumber,lincr,ldecr,pdecr, pincr, ptq,gai, pastMaxAI, kl,aPTQ, phylPTQ1, p,choosePhyllUse);
        System.out.println(" phyllochron: "+res0.phyllochron+" pastMaxAI: "+res0.pastMaxAI);

        System.out.println(((new BigDecimal(res0.pastMaxAI)).setScale(2, BigDecimal.ROUND_HALF_DOWN)).equals((new BigDecimal(0)).setScale(2, BigDecimal.ROUND_HALF_DOWN)));
        System.out.println(((new BigDecimal(res0.phyllochron)).setScale(2, BigDecimal.ROUND_HALF_DOWN)).equals((new BigDecimal(36.48)).setScale(2, BigDecimal.ROUND_HALF_DOWN)));
    }
}
